From 9e9278a7239c5c754859d48f3368a841ad08ff22 Mon Sep 17 00:00:00 2001
From: Hubert CHAUMETTE <hchaumette@esoftthings.com>
Date: Fri, 10 Apr 2020 07:31:55 +0000
Subject: [PATCH] r8a77980.dtsi: Add spi-mfis

Add SPI/MFIS device nodes and compatibility with spidev driver.
---
 arch/arm64/boot/dts/renesas/r8a77980.dtsi | 30 ++++++++++++++++++++++++++++++
 drivers/spi/spidev.c                      |  1 +
 2 files changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a77980.dtsi b/arch/arm64/boot/dts/renesas/r8a77980.dtsi
index becdfe1..63d2609 100755
--- a/arch/arm64/boot/dts/renesas/r8a77980.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77980.dtsi
@@ -1654,4 +1654,34 @@
 				      <&gic GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(4) |
 				       IRQ_TYPE_LEVEL_LOW)>;
 	};
+
+	spi-mfis {
+		compatible = "spi-mfis";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		spidev@0 {
+			compatible = "spi-mfis";
+			spi-max-frequency = <1000000>;
+			reg = <0>;
+		};
+
+		spidev@1 {
+			compatible = "spi-mfis";
+			spi-max-frequency = <1000000>;
+			reg = <1>;
+		};
+
+		spidev@2 {
+			compatible = "spi-mfis";
+			spi-max-frequency = <1000000>;
+			reg = <2>;
+		};
+
+		spidev@3 {
+			compatible = "spi-mfis";
+			spi-max-frequency = <1000000>;
+			reg = <3>;
+		};
+	};
 };
diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 788d2d6..9df898d 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -670,6 +670,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "ge,achc" },
 	{ .compatible = "semtech,sx1301" },
 	{ .compatible = "spi-gpio" },
+	{ .compatible = "spi-mfis" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.7.4

